File: stepik-union/src/main/java/org/stepik/core/courseFormat/StudyNodeComparator.java
package org.stepik.core.courseFormat;

import java.util.Comparator;

/**
 * @author meanmail
 */
class StudyNodeComparator implements Comparator<StudyNode> {

    private static Comparator<? super StudyNode> instance;

    public static Comparator<? super StudyNode> getInstance() {
        if (instance == null) {
            instance = new StudyNodeComparator();
        }
        return instance;
    }

    @Override
    public int compare(StudyNode item1, StudyNode item2) {
        return Integer.compare(item1.getPosition(), item2.getPosition());
    }
}

File: stepik-union/src/main/java/org/stepik/core/courseFormat/StudyNodeComparator.kt
package org.stepik.core.courseFormat

import java.util.*


internal class StudyNodeComparator : Comparator<StudyNode<*, *>> {

    override fun compare(item1: StudyNode<*, *>, item2: StudyNode<*, *>): Int {
        return Integer.compare(item1.position, item2.position)
    }

    companion object {

        val instance: Comparator<in StudyNode<*, *>> by lazy {
            StudyNodeComparator()
        }
    }
}
